src/data/multivariate_polynomial.lean

```diff
@@ -270,39 +271,193 @@ end
 
 end map
 
+section degrees
+
+section comm_semiring
+
+def degrees (p : mv_polynomial σ α) : multiset σ :=
+p.support.sup (λs:σ →₀ ℕ, s.to_multiset)
```

Comment:
It's not obvious to me that this definition is correct (useful for something). It would say that the degree of `x^2 * y + y^2 * x` is `[2,2]`, which looks like total degree `4` even though it is degree `3` homogeneous. What's the purpose of this?

Reply:
We restrict the overall dimension for each variable. Instead of using `degrees_of` I chose to use multisets, which behaved nicer in many ways. The main application is found in `field_theory.mv_polynomial` where it is used to bound the degrees of the indicator function. `field_theory.mv_polynomial` shows that a function over a finite field can be represented as a multivariate polynomial with restricted degree.